src/main/java/com/github/susom/dbgoodies/etl/BigQueryWriter.java

```diff
@@ -320,11 +321,15 @@ private Table checkTable(BigQuery bigquery){
       }
     }
 
-    return InsertAllRequest.RowToInsert.of(getRowId(row), row);
+    if(entryIdFields!=null && entryIdFields.length>0)
+      return InsertAllRequest.RowToInsert.of(getRowId(row), row);
+    else
+      return InsertAllRequest.RowToInsert.of(row);
+
```

Comment:
Should always use the braces for if/else...

Reply:
fixed in the latest commit.